lib/ui/widgets/gallery.dart

```diff
@@ -0,0 +1,339 @@
+  late final AnimationController unlikeController;
+  late final Animation<double> likeAnimation;
+  late final Animation<double> unlikeAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    controller = PageController(initialPage: widget.initialPage);
+
+    likeController = AnimationController(
+      duration: const Duration(milliseconds: 500),
+      vsync: this,
+      upperBound: 0.8,
+    )..addStatusListener((status) {
+        if (status == AnimationStatus.completed) likeController.reset();
+      });
+    unlikeController = AnimationController(
+      duration: const Duration(milliseconds: 500),
+      vsync: this,
+      upperBound: 0.8,
+    )..addStatusListener((status) {
+        if (status == AnimationStatus.completed) unlikeController.reset();
+      });
+
+    unlikeAnimation = CurvedAnimation(
+      parent: unlikeController,
+      curve: Curves.elasticOut,
+    );
+    likeAnimation = CurvedAnimation(
+      parent: likeController,
+      curve: Curves.elasticOut,
+    );
+  }
+
+  Future<void> _downloadImage(Uri url) async {
+    final messenger = ScaffoldMessenger.of(context);
+    try {
+      final response = await http.get(url);
+      if (response.statusCode != 200) throw Exception();
+      final baseTempDir = await getTemporaryDirectory();
+      final tempDir = await baseTempDir.createTemp();
+      final tempFile = File(
+        '${tempDir.path}/${url.pathSegments.last}',
+      );
+      await tempFile.writeAsBytes(response.bodyBytes);
+      await GallerySaver.saveImage(tempFile.path);
+
+      messenger.showSnackBar(
+        const SnackBar(
+          behavior: SnackBarBehavior.floating,
+          content: Text('Succesfully saved the image.'),
+        ),
+      );
+    } catch (_) {
+      messenger.showSnackBar(
+        const SnackBar(
+          behavior: SnackBarBehavior.floating,
+          content: Text('Could not download the image.'),
+        ),
+      );
+    }
+  }
+
+  Future<void> _shareImage(Uri url) async {
+    final messenger = ScaffoldMessenger.of(context);
+    try {
+      final response = await http.get(url);
+      if (response.statusCode != 200) throw Exception();
+      final file = XFile.fromData(
+        response.bodyBytes,
+        mimeType: lookupMimeType(url.path, headerBytes: response.bodyBytes),
+        name: url.pathSegments.last,
+      );
+      await Share.shareXFiles([file]);
+    } catch (_) {
+      messenger.showSnackBar(
+        const SnackBar(
+          behavior: SnackBarBehavior.floating,
+          content: Text('Could not share the image.'),
+        ),
+      );
+    }
+  }
+
+  Future<void> _likePhoto(List<AlbumPhoto> photos, int index) async {
+    final messenger = ScaffoldMessenger.of(context);
+    if (photos[index].liked) {
+      unlikeController.forward();
+    } else {
+      likeController.forward();
+    }
+    try {
+      await BlocProvider.of<LikedPhotosCubit>(context).updateLike(
```

Comment:
This breaks the gallery on the normal album view right? because it would try using the liked photos cubit that doesn't exist there

Reply:
I'll push a fix in a bit